File: client/src/app/classes/room-data.ts
import { Player } from './player';
import { Spectator } from './spectator';

export class RoomData {
    name: string;
    timeTurn: string;
    passwd: string;
    players: Player[];
    spectators: Spectator[];

    constructor(name: string, timeTurn: string, passwd: string, players: Player[], spectators: Spectator[]) {
        this.name = name;
        this.timeTurn = timeTurn;
        this.passwd = passwd;
        this.players = players;
        this.spectators = spectators;
    }
}

File: client_leger/lib/models/room-data.dart
import 'package:client_leger/models/player.dart';
import 'package:client_leger/models/spectator.dart';

class RoomData {
  String name;
  String timeTurn;
  bool isBonusRandom;
  String passwd;

  List<Player>players;

  List<Spectator> spectators;

  RoomData(
      {required this.name, required this.timeTurn, required this.isBonusRandom, required this.passwd, required this.players, required this.spectators});
}
